File: ERP/HomeFloory/Entities/Korpa.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace HomeFloory.Models;

public partial class Korpa
{
    public decimal IdKorpa { get; set; }

    public decimal? CenaDostave { get; set; }

    [DefaultValue(0)]
    public decimal? UkupnaCena { get; set; }

    public DateTime? Datum { get; set; }

    public string? Status { get; set; }

    public string? PaymentIntent { get; set; }

    public string? ClientSecret { get; set; }

    [DefaultValue(1)]
    public decimal IdPlacanje { get; set; }

    public decimal IdDostava { get; set; }

    public virtual ICollection<DodatiProizvodi> DodatiProizvodi { get; set; } = new List<DodatiProizvodi>();

    public virtual Dostava? IdDostavaNavigation { get; set; } = null!;

    public virtual Placanje? IdPlacanjeNavigation { get; set; } = null!;
}

File: Frontend/src/app/shared/models/korpa.ts
import { AdresaIsporuke } from "./korisnik"

export interface DodatiProizvodi {
    idDodatiProizvodi: number
    idProizvod: number
    idKorpa: number
    cena: number
    kolicina: number
    urlSlike?: string
    naziv?: string
    idProizvodNavigation?: IdProizvodNavigation
}


export interface Korpa {
    idKorpa: number
    cenaDostave: number
    ukupnaCena: number
    status?: string
    datum?: Date
    paymentIntent?: string
    clientSecret?: string
    idKorisnik?: number
    idDostava: number
    dodatiProizvodi: DodatiProizvodi[]
    idDostavaNavigation?: IdDostavaNavigation
    idPlacanjeNavigation?: IdPlacanjeNavigation
}

export interface Porudzbina {
    cenaDostave: number
    ukupnaCena: number
    idDostava: number
    idKorisnik: number
    dodatiProizvodi?: DodatiProizvodi[]
}

export interface IdDostavaNavigation {
    idDostava: number
    tipDostave: string
    nazivSluzbe: string
    cenaUsluge: number
    rokDostave: string
    korpe: any[]
}

export interface IdPlacanjeNavigation {
    idPlacanje: number
    status: string
    datum: string
    idKorisnik: number
    idKorisnikNavigation: any
    korpe: any[]
}

export interface IdProizvodNavigation {
    idProizvod: number
    naziv: string
    opis: string
    kolicinaNaStanju: number
    cenaPoM2: number
    paketPoM2: number
    dimenzija: string
    nijansa: string
    urlSlike: string
    idKategorija: number
    idProizvodjac: any
    dodatiProizvodi: DodatiProizvodi[]
    idKategorijaNavigation: any
    idProizvodjacNavigation: any
}


export interface IdKorpaNavigation {
    idKorpa: string
    cenaDostave: number
    ukupnaCena: number
    idPlacanje: number
    idDostava: number
    dodatiProizvodi: any[]
    idDostavaNavigation: any
    idPlacanjeNavigation: any
}

export class Korpa implements Korpa {
    idKorpa: number = Math.floor(Math.random() * 100) + 2;
    dodatiProizvodi: DodatiProizvodi[] = [];
}

export interface BasketTotal {
    shipping: number
    subtotal: number
    total: number
}